app/assets/javascripts/pageflow/editor/views/uploader_view.js

```diff
@@ -14,51 +10,42 @@ pageflow.UploaderView = Backbone.Marionette.View.extend({
 
     this.bindUIElements();
 
-    this.$el.fileupload({
-      acceptFileTypes: new RegExp ('(\\.|\\/)(bmp|gif|jpe?g|png|ti?f|wmv|mp4|mpg|mov|asf|asx|avi|' +
-                                   'm?v|mpeg|qt|3g2|3gp|3ivx|divx|3vx|vob|flv|dvx|xvid|mkv|vtt)$',
-                                   'i'),
-      dataType: 'json',
-
+    var form = this.$el;
+    var fileInput = this.$('input:file');
+    fileInput.fileupload({
+      fileInput: fileInput,
+      type: 'POST',
+      autoUpload: false,
+      paramName: 'file',
+      dataType: 'XML',
+      replaceFileInput: false,
+      acceptFileTypes: new RegExp('(\\.|\\/)(bmp|gif|jpe?g|png|ti?f|wmv|mp4|mpg|mov|asf|asx|avi|' +
+        'm?v|mpeg|qt|3g2|3gp|3ivx|divx|3vx|vob|flv|dvx|xvid|mkv|vtt)$',
+        'i'),
       add: function(event, data) {
-        try {
-          pageflow.fileUploader.add(data.files[0]).then(function(record) {
-            data.record = record;
-            var xhr = data.submit();
-
-            that.listenTo(data.record, 'uploadCancelled', function() {
-              xhr.abort();
-            });
+        pageflow.fileUploader.add(data.files[0]).then(function (record) {
+          data.record = record;
+
+          var s3UplopadConfig = record.attributes.s3_direct_upload_config;
+          form.data({host: s3UplopadConfig.host});
+          fileInput.fileupload({
+            url: s3UplopadConfig.url,
+            formData: s3UplopadConfig.fields
           });
-        }
-        catch(e) {
-          if (e instanceof pageflow.UploadError) {
-            pageflow.app.trigger('error', e);
-          }
-          else {
-            throw(e);
-          }
-        }
```

Comment:
The `try`/`catch` block is still needed to turn [`FileUploader` exceptons](https://github.com/codevise/pageflow/blob/1f7c77d3da2978a9b67e60182f401cebcc83e199/app/assets/javascripts/pageflow/editor/models/file_uploader.js#L25) into [alerts](https://github.com/codevise/pageflow/blob/1f7c77d3da2978a9b67e60182f401cebcc83e199/app/assets/javascripts/pageflow/editor/initializers/error_listener.js#L2).